packages/toolkit/src/tests/combineSlices.typetest.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-lone-blocks */
+import type { Reducer, Slice, WithSlice } from '@reduxjs/toolkit'
+import { combineSlices } from '@reduxjs/toolkit'
+import { expectExactType, expectType } from './helpers'
+
+declare const fooSlice: Slice<true, {}, 'foo'>
+
+declare const barSlice: Slice<true, {}, 'bar'>
+
+declare const bazReducer: Reducer<true>
```

Comment:
Maybe give these different types, `boolean`, `string` and `number` or something to make sure that not all slices in the end accidentally get the type from one slice/reducer.